docs/guides/cache/registry.md

```diff
@@ -0,0 +1,106 @@
+> can act as a simple drop-in replacement):
+>
+> ```console
+> docker buildx create --use --driver=docker-container
+> ```
+
+To import and export your cache using the `registry` storage backend we use the
+`--cache-to` and `--cache-from` flags and point it to our desired image target
+using the `ref` parameter:
+
+```console
+$ docker buildx build --push -t <user>/<image> \
+  --cache-to type=registry,ref=<user>/<cache-image> \
+  --cache-from type=registry,ref=<user>/<cache-image> .
+```
+
+You can choose any valid reference value for `ref`, as long as it's not the
+same as the target location that you push your image to. You might choose
+different tags (e.g. `foo/bar:latest` and `foo/bar:build-cache`), separate
+image names (e.g. `foo/bar` and `foo/bar-cache`), or even different
+repositories (e.g. `docker.io/foo/bar` and `ghcr.io/foo/bar`).
+
+If the cache does not exist, then the cache import step will fail, but the
+build will continue.
+
+## Cache options
+
+Unlike the simple `inline` cache, the `registry` cache has lots of parameters to
+adjust its behavior.
+
+### Cache mode
+
+Build cache can be exported in one of two modes: `min` or `max`, with either
+`mode=min` or `mode=max` respectively. For example, to build the cache with
+`mode=max`:
+
+```console
+$ docker buildx build --push -t <user>/<image> \
+  --cache-to type=registry,ref=<user>/<cache-image>,mode=max \
+  --cache-from type=registry,ref=<user>/<cache-image> .
+```
+
+Note that only `--cache-to` needs to be modified, as `--cache-from` will
+automatically extract the relevant parameters from the resulting output.
+
+In `min` cache mode (the default), only layers that are exported into the
+resulting image are cached, while in `max` cache mode, all layers are cached,
+even those of intermediate steps.
+
+While `min` cache is typically smaller (which speeds up import/export times,
+and reduces storage costs), `max` cache is more likely to get more cache hits.
+Depending on the complexity and location of your build, you should experiment
+with both parameters to get the results.
+
+### Cache compression
+
+Since `registry` cache is exported separately from the main build result, you
+can specify separate compression parameters for it (which are similar to the
+options provided by the `image` exporter). While the defaults have been
+selected to provide a good out-of-the-box experience, you may wish to tweak the
+parameters to optimize for storage vs compute costs.
+
+To select the compression algorithm, you can use the `compression=<uncompressed|gzip|estargz|zstd>`
+option. For example, to build the cache with `compression=zstd`:
+
+```console
+$ docker buildx build --push -t <user>/<image> \
+  --cache-to type=registry,ref=<user>/<cache-image>,compression=zstd \
+  --cache-from type=registry,ref=<user>/<cache-image> .
+```
+
+The `compression-level=<value>` option can be used alongside the `compression`
```

Comment:
if this option was not specified, what was the default `compression-level` value for `estargz` and `zstd`?

Reply:
These are properties of the underlying libraries we use - for `gzip`/`estargz` this is `compress/gzip`, while for `zstd` this is `github.com/klauspost/compress/zstd`. Since these constants could change at any time, I don't think we should list them here.

Unfortunately, not sure if we should link to upstream for these: https://pkg.go.dev/compress/gzip doesn't have neat docs on what the default is, though https://github.com/klauspost/compress/tree/master/zstd#status does (under the hood it doesn't use the same numbering level for zstd though).

I'll put the ranges of valid values in, and add links to the projects that do the compression, so users can at least find this info.